DatadogLogs/Sources/Log/SynchronizedAttributes.swift

```diff
@@ -0,0 +1,47 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+import DatadogInternal
+
+/// A thread-safe container for managing attributes in a key-value format.
+/// This class allows concurrent access and modification of attributes, ensuring data consistency
+/// through the use of a `ReadWriteLock`. It is designed to be used in scenarios where attributes
+/// need to be safely managed across multiple threads or tasks.
+internal final class SynchronizedAttributes: Sendable {
+    /// The underlying dictionary of attributes, wrapped in a `ReadWriteLock` to ensure thread safety.
+    private let attributes: ReadWriteLock<[String: Encodable]>
```

Comment:
/question We are not using the `ReadWriteLock` as a property wrapper in multiple places, is it because it would require it to be a `var`? Can't this be solved by making `ReadWriteLock` Sendable?

Reply:
The `ReadWriteLock` is [`Sendable` already](https://github.com/DataDog/dd-sdk-ios/blob/55ded08837526cabafe55d1bb3af77caacd0e7ae/DatadogInternal/Sources/Concurrency/ReadWriteLock.swift#L16) - it was done in [this PR](https://github.com/DataDog/dd-sdk-ios/pull/1778). And yes, if using property wrapper, this will become `var`, raising concurrency issue. I agree with what was [discussed here](https://github.com/DataDog/dd-sdk-ios/pull/1778#discussion_r1568741596) - property wrappers become quite unusable with strict concurrency.

Reply:
Yes, makes sense 👍